frontend_server/server.js

```diff
@@ -69,6 +69,11 @@ const apiProxyOptions = {
   target: assertEnvVar("DATA_SERVER_URL"),
   changeOrigin: true, // needed for virtual hosted sites
   pathRewrite: { '^/api': '' },
+  onProxyReq: (proxyReq, req, res) => {
```

Comment:
@aaronsn This feels pretty hacky. Is there a better convention for storing the token until the onProxyReq event?

Cant actually fetch the token during the event because the proxy request will be sent before the token is fetched.

Reply:
Hmm, I see. Did you test it out and it works this way?

I'm guessing there's probably a way to create a custom middleware function that waits for the header and then calls the proxy directly with the header as an argument..... buuut we're definitely reaching the limit of my understanding of express.

TBH this doesn't seem that bad to me. It's probably less complex than anything else I can think of. I'd maybe rename `AuthorizationTemp` to something like `Authorization_DataServer` and expand on the comment with why it's set up this way.

Reply:
Yea, this is the only way I could get it to work. I tried writing additional middleware to only use basic auth on routes that didn't start with /api but no matter what I did the basic auth header still ended up there.